packages/general-settings/middleware.js

```diff
@@ -1,12 +1,34 @@
 import { getURL } from '@bufferapp/publish-formatters';
+import { actionTypes as profileActionTypes } from '@bufferapp/publish-profile-sidebar';
+import {
+  actions as asyncDataFetch,
+  actions as dataFetchActions,
+} from '@bufferapp/async-data-fetch';
 import { actionTypes } from './reducer';
 
-export default () => next => (action) => {
+export default ({ dispatch }) => next => (action) => {
   next(action);
   switch (action.type) {
     case actionTypes.SET_DIRECT_POSTING:
       window.location = getURL.getInstagramDirectPostingURL(action.profileId);
       break;
+    case profileActionTypes.SELECT_PROFILE:
+      dispatch(dataFetchActions.fetch({
+        name: 'getLinkShortener',
+        args: {
+          profileId: action.profile.id,
```

Comment:
Can you get the `profileId` directly out of action here, like:
```suggestion
        profileId: action.profileId
```

Reply:
Hey Lola, great question here, I had a look and I think that since this is triggering from the SELECT_PROFILE action, it looks like this is the place in the data that the id is stored, and seems consistent with other places in our codebase where this is being accessed... with that being said, its quite a difficult challenge to work out where to source data as there is no definition of the data objects we are seeing, so it makes it quite challenging to find the right piece of data here, so thankyou for mentioning it, as it meant i could double check 👍